tools/local-environment/tests/watcherImproperUpdate.ts

```diff
@@ -0,0 +1,140 @@
+import { CoreContracts } from "@nomad-xyz/sdk/nomad";
+import { ethers } from "ethers";
+import { LocalNetwork, Nomad, Key, utils } from "../src";
+
+import { sleep } from "../src/utils";
+
+async function improperUpdateCase(homeOrReplica: string) {
+    let success = false;
+
+  const tom = new LocalNetwork("tom", 1000, "http://localhost:9545");
+  const jerry = new LocalNetwork("jerry", 2000, "http://localhost:9546");
+
+  const updaterKey = new Key();
+  const watcherKey = new Key();
+  const jerryDeployerKey = new Key();
+  const jerrySignerKey = new Key();
+  const tomDeployerKey = new Key();
+  const tomSignerKey = new Key();
+
+  tom.addKeys(updaterKey, watcherKey, tomDeployerKey, tomSignerKey);
+  jerry.addKeys(updaterKey, watcherKey, jerryDeployerKey, jerrySignerKey);
+
+  await Promise.all([tom.up(), jerry.up()]);
+
+  console.log(`Started both`);
+
+  const n = new Nomad(tom);
+  n.addNetwork(jerry);
+
+  n.setUpdater(tom, updaterKey);
+  n.setWatcher(tom, watcherKey);
+  n.setDeployer(tom, tomDeployerKey);
+  n.setSigner(tom, tomSignerKey);
+
+  n.setUpdater(jerry, updaterKey); // Need for an update like updater
+  n.setWatcher(jerry, watcherKey); // Need for the watcher
+  n.setDeployer(jerry, jerryDeployerKey); // Need to deploy all
+  n.setSigner(jerry, jerrySignerKey); // Need for home.dispatch
+
+  await n.deploy({ injectSigners: true });
```

Comment:
seems to me like everything before this line should be extracted to a generic fn for standing up 2 full nomad tom and jerry deploys + agents - no?

Reply:
it is not always the case, but in most, so I agree